### PR TITLE
Fix X-Frame enabled behaviour

### PR DESCRIPTION
**Pull request by potiuk**
The #19491 incorrectly changed condition on assigning the
X-Frame-Options header DENY. It actually was not possible to set
the DENY header.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
@uranusjr  - we both approved the change  in #19491  but I think we were both wrong as we have not looked beyond just the diff :) (and it did not work before either actually)

File: airflow/www/extensions/init_security.py [hunk 1]
```diff
@@ -31,12 +31,11 @@ def init_xframe_protection(app):
     See also: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
     """
     x_frame_enabled = conf.getboolean('webserver', 'X_FRAME_ENABLED', fallback=True)
-    if not x_frame_enabled:
+    if x_frame_enabled:
         return
```
Comment (ashb) [inline airflow/www/extensions/init_security.py:35]: Hmmmm, does "correcting" this now end up breaking things I wonder?

i.e. do we follow the what our docs say, or do we update the docs to match what the implementation currently does?

Comment (ashb) [reply airflow/www/extensions/init_security.py:35]: Oh this only landed in 2.2.5, right.

Comment (potiuk) [reply airflow/www/extensions/init_security.py:35]: Pre 2.2.5 it was actually worse because the flag work opposite of its description.

Comment (ashb) [reply airflow/www/extensions/init_security.py:35]: Obligatory XKCD https://xkcd.com/1172/

Comment (potiuk) [reply airflow/www/extensions/init_security.py:35]: Very, very much this.

Comment (ashb) [reply airflow/www/extensions/init_security.py:35]: Anyway -- I think this warrants a news fragment please.

Comment (potiuk) [reply airflow/www/extensions/init_security.py:35]: Absolutely :). The pleasure is mine. But I am not entirely sure what to write :).

"Finally it works as expected" :D ?

Comment (potiuk) [reply airflow/www/extensions/init_security.py:35]: I made an attempt to describe it - but it's a bit wordy :).

**Review by ashb — APPROVED** [review]
